### PR TITLE
Code clean up to to fix issue #234 - ConnectionsState exposes a setter into internal state

### PR DESCRIPTION
**Pull request by ya-pulser**

### CODE REVIEW

**Comment by artemredkin** [comment]
@swift-server-bot test this please

File: Sources/AsyncHTTPClient/ConnectionsState.swift [hunk 1]
```diff
@@ -15,29 +15,29 @@
...
-        case fail(Waiter, Error)
-        indirect case closeAnd(Connection, Action)
-        indirect case parkAnd(Connection, Action)
+        case fail(Waiter<ConnectionType>, Error)
+        indirect case closeAnd(ConnectionType, Action<ConnectionType>)
+        indirect case parkAnd(ConnectionType, Action<ConnectionType>)
     }
 
-    struct ConnectionsState {
+    struct ConnectionsState<ConnectionType: PoolManageableConnection> {
```
Comment (artemredkin) [inline Sources/AsyncHTTPClient/ConnectionsState.swift:30]: small comment from me: this generic is only needed to we can mock the `Connection` type in order to make sure that `ConnectionState` tests do not depend on `HTTP1ConnectionProvider`.

**Review by artemredkin — APPROVED** [review]

**Comment by artemredkin** [comment]
@swift-server-bot test this please

**Comment by Lukasa** [comment]
@swift-server-bot add to whitelist

**Review by Lukasa — APPROVED** [review]
Nice, I'm entirely happy with this refactor, great job.

**Pull request merged**